src/Database/Models/Revision.php

```diff
@@ -34,7 +34,7 @@ public function getNewValueAttribute($value)
      */
     public function getOldValueAttribute($value)
     {
-        if ($this->cast == 'date' && !is_null($value)) {
+        if ($this->getAttribute('cast') === 'date' && !is_null($value)) {
```

Comment:
Why was this required?

Reply:
It fixes an issue where the `cast` property might not be set.

Reply:
Wouldn't that just cycle through the getAttribute() logic anyways by way of the `__get()` magic method?

Reply:
It does, but where possible, we should avoid relying on magic in internal methods as it's not easily deducible. Userland code may use magic.

Reply:
Hmm, I'm not sure. It would change the behaviour if someone had specified a `cast` property on the model - in these sorts of cases I feel like it's better to use the standard way of interacting with model attributes rather than bypass one of the layers.

Reply:
Sorry @LukeTowers, just circling back to this only now. I'm not sure what you mean here though - both using this `getAttribute` call or using the `__get` magic method do the same thing, so what behaviour would be changed here?

Reply:
@bennothommo I mean that if someone defines `$cast` as a property on a class that extends this model class then this logic would no longer get that property value. However, this is unlikely so I no longer care about this change :P